lib/app/modules/quality_reporting/measure_availability_sequence.rb

```diff
@@ -21,9 +21,21 @@ class MeasureAvailability < SequenceBase
           link 'https://www.hl7.org/fhir/measure.html'
           desc 'Check to make sure specified measure is available on test server'
         end
+
+        # Look for matching measure from cqf-ruler datastore by resource id
         measure_id = @instance.measure_to_test
-        query_response = @client.search(FHIR::Measure, search: { parameters: { _id: measure_id } })
+        measure_resource = @instance.module.measures.find { |m| m.resource.id == measure_id }
+
+        @client.additional_headers = { 'x-api-key': @instance.api_key, 'Authorization': @instance.auth_header }
+
+        # Search system for measure by identifier and version
+        measure_identifier = measure_resource.resource.identifier.find { |id| id.system == 'http://hl7.org/fhir/cqi/ecqm/Measure/Identifier/cms' }
+        measure_version = measure_resource.resource.version
+        query_response = @client.search(FHIR::Measure, search: { parameters: { identifier: measure_identifier.value, version: measure_version } })
```

Comment:
This line errors out when testing `measure-exm165-FHIR3` with the preloaded cqf-ruler. The error is on the `.value` in the block.

```
undefined method `value' for nil:NilClass
```

Reply:
I would argue that the identifier is what should be seen in the dropdown populated by our cqf-ruler. Measures in cqf-ruler without identifiers shouldn't show up in the dropdown. Or perhaps show up with a warning that they have no identifier.

Reply:
Really what we want is both the `identifier` and `version` of the measure in the dropdown, since that's what we're keying on in our search. Given how this currently works with the erb form input, it was easier to just use a single value (the `id`) for this purpose, and since we control the IDs for our embedded CQF Ruler, we can ensure those IDs contain both pieces of information.

I agree measures without `identifier` shouldn't appear in the dropdown, so I'll make that change.

Reply:
Actually I just looked into it and I think it's pretty simple in the erb to put what we want in the actual dropdown while still using the identifier as the actual value, so I'll do that